schema/deploy/warehouse/identifier/triggers/barcode-distance-check.sql

```diff
@@ -27,15 +27,16 @@ create or replace function warehouse.identifier_barcode_distance_check() returns
         -- identifier table but serializes other invocations of this function.
         lock table warehouse.identifier in exclusive mode;
 
-        select array_agg(barcode) into strict conflicting_barcodes from (
+        select barcode into conflicting_barcode from (
             select barcode
               from warehouse.identifier
              where hamming_distance_ci(barcode, NEW.barcode) < minimum_distance
             except
             select old_barcode
+             limit 1
```

Comment:
Hmm, I'm now wondering if this is actually correct or not. The `limit 1` may not work as intended depending on how `except` is applied. Looking into this more.

Reply:
I had the same thought, and I saw it working. Let me know what you find.

Reply:
Ok, it does work as intended (assuming this is a valid test):

```sql
tom@[local]/tom=# values ('a'), ('b'), ('c') except values ('b') limit 1;
 column1 
─────────
 a
(1 row)

tom@[local]/tom=# values ('a'), ('b'), ('c') except values ('a') limit 1;
 column1 
─────────
 c
(1 row)

tom@[local]/tom=# values ('a'), ('b'), ('c') except values ('b') limit 1;
 column1 
─────────
 a
(1 row)

tom@[local]/tom=# values ('a'), ('b'), ('c') except values ('c') limit 1;
 column1 
─────────
 a
(1 row)

tom@[local]/tom=# values ('x'), ('x'), ('x') except values ('x') limit 1;
 column1 
─────────
(0 rows)

```